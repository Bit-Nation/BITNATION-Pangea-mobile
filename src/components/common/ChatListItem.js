// @flow

/**
 * @desc Component that renders chat list item.
 * @type React.Component
 */

import React from 'react';
import { Image } from 'react-native';

import ListItem from './ListItem';
import GlobalStyles from '../../global/Styles';

type Props = {
  /**
  * @desc Name of chat partner.
  */
  name: string,
  /**
   * @desc Last message to show on preview.
   */
  lastMessage?: string | null,
  /**
   * @desc Base64 avatar of partner.
   */
  avatar: Image.propTypes.source,
  /**
   * @desc Id that will be passed in onPress callback.
   */
  id: string,
  /**
   * @desc Callback on press item.
   * @param Id of item that was pressed.
   */
  onPress: (id: string) => void,
}

const ChatListItem = ({
  name, id, onPress, avatar, lastMessage,
}: Props) => (
  <ListItem
    id={id}
    text={name}
    textStyle={GlobalStyles.chatTitleStyle}
    onPress={onPress}
    iconSource={avatar}
    style={GlobalStyles.itemWithSubtitle}
    subtitle={lastMessage}
  />
);

ChatListItem.defaultProps = {
  partnerName: '',
  onPress: () => null,
  partnerAvatar: null,
  lastMessage: undefined,
};

export default ChatListItem;
