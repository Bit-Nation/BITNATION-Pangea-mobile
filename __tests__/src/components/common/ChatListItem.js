// @flow

import React from 'react';
import { shallow } from 'enzyme';

import ChatListItem from '../../../../src/components/common/ChatListItem';

describe('ChatListItem tests', () => {
  describe('Rendering', () => {
    test('With text', () => {
      const wrapper = shallow(<ChatListItem
        id={1}
        name='Test list item'
        lastMessage='Some message'
        unreadMessages={false}
        onPress={jest.fn()}
      />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour', () => {
    test('Press', () => {
      const mockFunc = jest.fn();
      const id = 1;
      const wrapper = shallow(<ChatListItem
        unreadMessages={false}
        name='Test list item'
        onPress={mockFunc}
        id={id}
      />);
      expect(wrapper).toMatchSnapshot();
      const render = wrapper.dive();
      const touchables = render.find('[testIDJest="Touchable"]');
      expect(touchables).toHaveLength(1);
      touchables.props().onPress();
      expect(mockFunc).toHaveBeenCalledTimes(1);
      expect(mockFunc).toHaveBeenCalledWith(id);
    });
  });
});

