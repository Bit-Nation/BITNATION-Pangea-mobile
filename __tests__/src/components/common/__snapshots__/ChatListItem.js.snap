// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatListItem tests Behaviour Press 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatListItem
    avatar={null}
    id="Test id"
    lastMessage={undefined}
    name="Test list item"
    onPress={[MockFunction]}
    unreadMessages={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "AdditionalLeftView": [Function],
      "AdditionalView": [Function],
      "disabled": false,
      "disclosureIconVisible": true,
      "iconSource": null,
      "id": "Test id",
      "onPress": [MockFunction],
      "style": Object {
        "height": 72,
      },
      "subtitle": undefined,
      "text": "Test list item",
      "textStyle": Object {
        "color": "#58595B",
        "fontFamily": "Roboto",
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0,
        "lineHeight": 22,
      },
      "value": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "AdditionalLeftView": [Function],
        "AdditionalView": [Function],
        "disabled": false,
        "disclosureIconVisible": true,
        "iconSource": null,
        "id": "Test id",
        "onPress": [MockFunction],
        "style": Object {
          "height": 72,
        },
        "subtitle": undefined,
        "text": "Test list item",
        "textStyle": Object {
          "color": "#58595B",
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 22,
        },
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ChatListItem tests Rendering With text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatListItem
    avatar={null}
    id="1"
    lastMessage="Some message"
    name="Test list item"
    onPress={[MockFunction]}
    unreadMessages={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "AdditionalLeftView": [Function],
      "AdditionalView": [Function],
      "disabled": false,
      "disclosureIconVisible": true,
      "iconSource": null,
      "id": "1",
      "onPress": [MockFunction],
      "style": Object {
        "height": 72,
      },
      "subtitle": "Some message",
      "text": "Test list item",
      "textStyle": Object {
        "color": "#58595B",
        "fontFamily": "Roboto",
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0,
        "lineHeight": 22,
      },
      "value": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "AdditionalLeftView": [Function],
        "AdditionalView": [Function],
        "disabled": false,
        "disclosureIconVisible": true,
        "iconSource": null,
        "id": "1",
        "onPress": [MockFunction],
        "style": Object {
          "height": 72,
        },
        "subtitle": "Some message",
        "text": "Test list item",
        "textStyle": Object {
          "color": "#58595B",
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 22,
        },
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
